### PR TITLE
WIP: Filterable record properties chart

### PR DESCRIPTION
**Pull request by bobbilee19**
WIP / Draft Mode: I'd like @thehenrytsai @diehuxx to cross check the way I've described these in a succinct way. And for the examples I've added for `contextId` and `parentId`

Fastest way to preview the chart I made: https://developer-tbd-website-site-git-issue-622-que-9e2fc2-tbd54566975.vercel.app/docs/web5/build/decentralized-web-nodes/query-from-dwn

### CODE REVIEW

**Comment by bobbilee19** [comment]
Context to add from spec:
- `schema` property, and if present its value MUST be a URI string that indicates the schema of the associated data.

- `recordId` property, and its value MUST be a [Computed Record ID](https://identity.foundation/decentralized-web-node/spec/#computed-record-ids).

- `contextId` property, and its value MUST be the deterministic ID for a contextually linked set of objects.

- `protocol` property, and its value MUST be a URI that denotes the Protocol an object is a part of.

- `parentId` property if the currently active entry for the record is a RecordsDelete or a CollectionWrite that has a declared a Commit Strategy. The object MUST NOT contain a `parentId` under any other circumstance. If present, the value of the `parentId` property MUST be the stringified [Version 1 CID](https://docs.ipfs.io/concepts/content-addressing/#identifier-formats) of the [DAG CBOR](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-cbor.md) encoded descriptor object of the previous RecordsWrite or RecordsDelete entry the message is intending to overwrite.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 1]
```diff
@@ -36,6 +36,21 @@ const { records } = await web5.dwn.records.query({
...
+| Property            | Value                                                |   Example   |
+| -----------         | ----------                                           | ----------- |  
+| **`attester`**      | Author DID of message                                | "did:example:bob" |
+| **`recipient`**     | Recipient DID of message                             | "did:example:alice"  |
+| **`protocol`**      | The URI of the protocol bucket in which to query     | "example.com" |
+| **`protocolPath`**  | Records under a protocol path across all context IDs | "example" |
+| **`contextId`**     | Message thread contextId                             | "12345678" |
+| **`schema`**        | The URI of the schema bucket in which to query       | "https://schema.org/Message" |
+| **`recordId`**      | Property contains the message recordId               | "aa36ec55-c59b-4f20-8143-10f74aac696d" |
+| **`parentId`**      | RecordsWrite message’s `entryId`                     | "91011123" |
```
Comment (diehuxx) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:50]:
```suggestion
| **`parentId`**      | `recordId` of the parent record in a protocol          | "bafyreianzpmhbgcgam5mys722vnsiuwn7y4ek6kjeyjptttquasw4hge2m" |
```

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 2]
```diff
@@ -36,6 +36,21 @@ const { records } = await web5.dwn.records.query({
...
 
+## Filterable Record Properties
+
+| Property            | Value                                                |   Example   |
+| -----------         | ----------                                           | ----------- |  
+| **`attester`**      | Author DID of message                                | "did:example:bob" |
+| **`recipient`**     | Recipient DID of message                             | "did:example:alice"  |
+| **`protocol`**      | The URI of the protocol bucket in which to query     | "example.com" |
+| **`protocolPath`**  | Records under a protocol path across all context IDs | "example" |
+| **`contextId`**     | Message thread contextId                             | "12345678" |
```
Comment (diehuxx) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:47]:
```suggestion
| **`contextId`**     | `recordId` of a root record of a protocol                             | 'bafyreianzpmhbgcgam5mys722vnsiuwn7y4ek6kjeyjptttquasw4hge2m'
 |
```

This is a tough one to express succinctly if it hasn't already been established that records within a protocol form a number of trees. Each `contextId` refers to one of those trees. When a new top-level "root" record is written, it starts a new tree. The `contextId` is `recordId` of the root record. All descendants of a given root record have the same `contextId`.

Personally, I would just assume the reader has some knowledge that dwn protocols form trees of records and that they learned this a previous page of documentation. If this isn't covered in our docs on protocols, we should update those.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 3]
```diff
@@ -36,6 +36,21 @@ const { records } = await web5.dwn.records.query({
 });
 ```
 
+## Filterable Record Properties
+
+| Property            | Value                                                |   Example   |
+| -----------         | ----------                                           | ----------- |  
+| **`attester`**      | Author DID of message                                | "did:example:bob" |
```
Comment (diehuxx) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:43]: Omit this one please. I looked through our code and we're not actually indexing on it, so the query is useless AFAICT. I'm either going to update the name or rip it out entirely.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 1 above)
Comment (diehuxx) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:50]: Also, I'm not sure how literal you mean to get with your example values. For `recordId`s, we're using [CIDs](https://docs.ipfs.tech/concepts/content-addressing/#cids-are-not-file-hashes), specifically CBOR + SHA256. I just used [this test fixture](https://github.com/TBD54566975/dwn-sdk-js/blob/main/tests/utils/test-data-generator.ts#L797-L802) to generate one randomly.

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 4]
```diff
@@ -36,6 +36,21 @@ const { records } = await web5.dwn.records.query({
 });
 ```
 
+## Filterable Record Properties
+
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:40]: Add a one-liner sentence here explaining what these are for

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 2 above)
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:47]: @bobbilee19 let's add a new issue to add this info to the Protocol doc and add a link here telling people to read more about it.

For this particular guide, let's incorporate Diane's suggestion minus the long string

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 5]
```diff
@@ -36,6 +36,21 @@ const { records } = await web5.dwn.records.query({
...
+| **`recipient`**     | Recipient DID of message                             | "did:example:alice"  |
+| **`protocol`**      | The URI of the protocol bucket in which to query     | "example.com" |
+| **`protocolPath`**  | Records under a protocol path across all context IDs | "example" |
+| **`contextId`**     | Message thread contextId                             | "12345678" |
+| **`schema`**        | The URI of the schema bucket in which to query       | "https://schema.org/Message" |
+| **`recordId`**      | Property contains the message recordId               | "aa36ec55-c59b-4f20-8143-10f74aac696d" |
+| **`parentId`**      | RecordsWrite message’s `entryId`                     | "91011123" |
+| **`dataFormat`**    | The IANA string for the data format to filter        | "application/json" |
+| **`dateCreated`**   | Date the record was created                          | "2023-04-30T22:49:37.713976Z" |
+
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:53]: Can we add an example code snippet using one of the filters just to show people where this shows up?

> **Code Sample**
> This snippet queries the DWN for records that have a parent record with a specific record ID:
> 
> ```js
> const response = await web5.dwn.records.query({
>   message: {
>     filter: {
>       parentId: 'bafyreianzpmhbgcgam5mys722vnsiuwn7y4ek6kjeyjptttquasw4hge2m',
>     }
>   }
> });
> ```

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 3 above)
Comment (bobbilee19) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:43]: Ok cool just wanted to make sure it was there before you removed it. But I'll remove it. Thanks!

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 2 above)
Comment (diehuxx) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:47]: How bout as a compromise, we use a truncated version of the real possible value i provided? eg “bafyreian…"

Comment (bobbilee19) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:47]: https://github.com/TBD54566975/developer.tbd.website/issues/651

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx (hunk 1 above)
Comment (bobbilee19) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx:50]: https://github.com/TBD54566975/developer.tbd.website/issues/651

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx [hunk 6]
```diff
@@ -36,6 +36,34 @@ const { records } = await web5.dwn.records.query({
...
+## Filterable Record Properties
+Here are the properties you can use to filter your record query, along with explanations and examples for each.
+
+| Property            | Value                                                |   Example   |
+| -----------         | ----------                                           | ----------- |  
+| **`recipient`**     | Recipient DID of message                             | "did:example:alice"  |
+| **`protocol`**      | The URI of the protocol bucket in which to query     | "example.com" |
+| **`protocolPath`**  | Records under a protocol path across all context IDs | "example" |
+| **`contextId`**     | `recordId` of a root record of a protocol            | "bafyreianzpmhbgcgam5mys722vnsiuwn..." |
+| **`schema`**        | The URI of the schema bucket in which to query       | "https://schema.org/Message" |
```
[the remark this answers is not in the document]
Comment (angiejones) [reply site/docs/web5/build/decentralized-web-nodes/query-from-dwn.mdx]: i fixed it

**Review by angiejones — APPROVED** [review]

**Pull request merged**